antenna/ext/crashpublish_base.py

```diff
@@ -0,0 +1,57 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import logging
+
+from everett.component import ConfigOptions, RequiredConfigMixin
+
+
+logger = logging.getLogger(__name__)
+
+
+class CrashPublishBase(RequiredConfigMixin):
+    """Crash publishstorage base class."""
+
+    required_config = ConfigOptions()
+
+    def __init__(self, config):
+        self.config = config.with_options(self)
+
+    def publish_crash(self, crash_id):
+        """Publish the crash id
+
+        FIXME(willkg): How should this method handle exceptions?
+
+        :arg str crash_id: The crash id as a string.
+
+        """
+        raise NotImplementedError
+
+
+class NoOpCrashPublish(CrashPublishBase):
+    """This is a no-op crash publish class that logs crashes it would have published.
+
+    It keeps track of the last 10 crash ids in ``.crashes`` instance attribute
+    with the most recently published crash id at the end of the list. This
+    helps when writing unit tests for Antenna.
```

Comment:
I didn't end up using this at all in the tests. I'll look into that. I either need to add some more test stuff, or take this out.